x-pack/solutions/security/plugins/security_solution/server/lib/siem_migrations/common/task/siem_migrations_task_runner.ts

```diff
@@ -0,0 +1,328 @@
+
+    try {
+      this.logger.debug('Initializing migration');
+      await this.withAbort(this.initialize());
+    } catch (error) {
+      migrationTaskTelemetry.failure(error);
+      if (error instanceof AbortError) {
+        this.logger.info('Abort signal received, stopping initialization');
+        return;
+      } else {
+        throw new Error(`Migration initialization failed. ${error}`);
+      }
+    }
+
+    const migrateItemTask = this.createMigrateItemTask(invocationConfig);
+    this.logger.debug(`Started translations. Concurrency is: ${TASK_CONCURRENCY}`);
+
+    try {
+      do {
+        const { data: migrationItems } = await this.data.items.get(migrationId, {
+          filters: { status: SiemMigrationStatus.PENDING },
+          size: TASK_BATCH_SIZE, // keep these items in memory and process them in the promise pool with concurrency limit
+        });
+        if (migrationItems.length === 0) {
+          break;
+        }
+
+        this.logger.debug(`Start processing batch of ${migrationItems.length} items`);
+
+        const { errors } = await initPromisePool<Stored<I>, void, Error>({
+          concurrency: TASK_CONCURRENCY,
+          abortSignal: this.abortController.signal,
+          items: migrationItems,
+          executor: async (migrationItem) => {
+            const itemTranslationTelemetry = migrationTaskTelemetry.startItemTranslation();
+            try {
+              await this.saveItemProcessing(migrationItem);
+
+              const migratedItem = await migrateItemTask(migrationItem);
+
+              await this.saveItemCompleted(migratedItem);
+              itemTranslationTelemetry.success(migratedItem);
+            } catch (error) {
+              if (this.abortController.signal.aborted) {
+                throw new AbortError();
+              }
+              itemTranslationTelemetry.failure(error);
+              await this.saveItemFailed(migrationItem, error);
+            }
+          },
+        });
+
+        if (errors.length > 0) {
+          throw errors[0].error; // Only AbortError is thrown from the pool. The task was aborted
+        }
+
+        this.logger.debug('Batch processed successfully');
+      } while (true);
+
+      migrationTaskTelemetry.success();
+      this.logger.info('Migration completed successfully');
+    } catch (error) {
+      await this.data.items.releaseProcessing(migrationId);
+
+      if (error instanceof AbortError) {
+        migrationTaskTelemetry.aborted(error);
+        this.logger.info('Abort signal received, stopping migration');
+      } else {
+        migrationTaskTelemetry.failure(error);
+        throw new Error(`Error processing migration: ${error}`);
+      }
+    } finally {
+      this.abort.cleanup();
+    }
+  }
+
+  /* Prepares the invokable task with raw input and output. */
+  async prepareTaskInvoke(input: P, config: RunnableConfig<C>) {
+    assert(this.task, 'Migration task is not defined');
+    return this.task(input, config);
+  }
```

Comment:
This does not _prepare_ the task, it just executes it. 
Consider changing the signature to something like:
```suggestion
  /* Executes the task with raw input and config, and returns the output promise. */
  async executeTask(input: P, config: RunnableConfig<C>): Promise<O> {
    assert(this.task, 'Migration task is not defined');
    return this.task(input, config);
  }
```

Reply:
Done.